Makefile

```diff
@@ -17,72 +17,57 @@
+	cl=`grep -n %changelog python-${PKGNAME}.spec | cut -d : -f 1` ; \
+	tail --lines=+$$(($$cl + 1)) python-${PKGNAME}.spec > speclog ; \
+	(head -n $$cl python-${PKGNAME}.spec ; echo "$$DATELINE" ; make --quiet changelog 2>/dev/null ; echo ""; cat speclog) > python-${PKGNAME}.spec.new ; \
+	mv python-${PKGNAME}.spec.new python-${PKGNAME}.spec ; rm -f speclog ; \
+	sed -i "s/Version:\( *\)$(VERSION)/Version:\1$$NEWVERSION/" python-${PKGNAME}.spec ; \
+	sed -i "s/version=\"$(VERSION)\"/version=\"$$NEWVERSION\"/" setup.py ; \
+	git add python-${PKGNAME}.spec setup.py ; \
+	git commit -m "New release: $$NEWVERSION"
 
+.PHONY: tag
 tag:
-	git tag -a -m "Tag as $(VERSION)" -f v$(VERSION)
-	@echo "Tagged as $(VERSION)"
-
-archive: check tag
-	git archive --format=tar --prefix=$(PKGNAME)-$(VERSION)/ $(VERSION) > $(PKGNAME)-$(VERSION).tar
-	gzip -9 $(PKGNAME)-$(VERSION).tar
-	@echo "The archive is in $(PKGNAME)-$(VERSION).tar.gz"
-
-release-pypi:
-	if ! $(PYTHON) setup.py sdist bdist_wheel; then \
-		echo ""; \
-		echo Distribution package build failed! Please verify that you have \'python3-wheel\' and \'python3-setuptools\' installed. >&2; \
-		exit 1; \
-	fi
-	if ! $(PYTHON) -m twine upload dist/*; then \
-		echo ""; \
-		echo Package upload failed! Make sure the \'twine tool\' is installed and you are registered >&2; \
-		exit 1; \
-	fi
+	git tag -a -m "Tag as $(VERSION)" -f $(TAG)
+	@echo "Tagged as $(TAG)"
 
-local:
-	@rm -rf $(PKGNAME)-$(VERSION).tar.gz
-	@rm -rf /tmp/$(PKGNAME)-$(VERSION) /tmp/$(PKGNAME)
-	@dir=$$PWD; cp -a $$dir /tmp/$(PKGNAME)-$(VERSION)
-	@cd /tmp/$(PKGNAME)-$(VERSION) ; $(PYTHON) setup.py -q sdist
-	@cp /tmp/$(PKGNAME)-$(VERSION)/dist/$(PKGNAME)-$(VERSION).tar.gz .
-	@rm -rf /tmp/$(PKGNAME)-$(VERSION)
-	@echo "The archive is in $(PKGNAME)-$(VERSION).tar.gz"
+.PHONY: push
+push:
+	git push --follow-tags
 
-rpmlog:
-	@git log --pretty="format:- %s (%ae)" $(VERSION).. |sed -e 's/@.*)/)/' | grep -v "Merge pull request"
-
-bumpver:
-	@NEWSUBVER=$$((`echo $(VERSION) |cut -d . -f 2` + 1)) ; \
-	NEWVERSION=`echo $(VERSION).$$NEWSUBVER |cut -d . -f 1,3` ; \
-	DATELINE="* `LC_ALL=C.UTF-8 date "+%a %b %d %Y"` `git config user.name` <`git config user.email`> - $$NEWVERSION-1"  ; \
-	cl=`grep -n %changelog python-${PKGNAME}.spec |cut -d : -f 1` ; \
-	tail --lines=+$$(($$cl + 1)) python-${PKGNAME}.spec > speclog ; \
-	(head -n $$cl python-${PKGNAME}.spec ; echo "$$DATELINE" ; make --quiet rpmlog 2>/dev/null ; echo ""; cat speclog) > python-${PKGNAME}.spec.new ; \
-	mv python-${PKGNAME}.spec.new python-${PKGNAME}.spec ; rm -f speclog ; \
-	sed -i "s/Version:   $(VERSION)/Version:   $$NEWVERSION/" python-${PKGNAME}.spec ; \
-	sed -i "s/version='$(VERSION)'/version='$$NEWVERSION'/" setup.py
+.PHONY: archive
+archive:
+	$(PYTHON) setup.py sdist bdist_wheel
+	@echo "The archive is in dist/$(PKGNAME)-$(VERSION).tar.gz"
 
-.PHONY: clean install tag archive local
+.PHONY: upload
+upload:
```

Comment:
I would recommend to change this to ``upload-pypi:``.

Reply:
I will keep it as it is for now.